src/main/java/org/mule/extension/sftp/internal/SftpInputStream.java

```diff
@@ -129,6 +129,35 @@ public InputStream get() {
       }
     }
 
+    private SftpFileAttributes getUpdatedStableAttributes(SftpConnector config, ConnectionManager connectionManager,
+                                                          SftpFileAttributes updatedAttributes)
+        throws ConnectionException {
+      SftpFileAttributes oldAttributes;
+      int retries = 0;
+      do {
+        retries++;
+        if (retries > MAX_SIZE_CHECK_RETRIES) {
+          break;
```

Comment:
you are already in a while, is better not to have two places that break the iteration. This should be part of the while condition.